pkg/system/filesys_windows.go

```diff
@@ -0,0 +1,62 @@
+// +build windows
+
+package system
+
+import (
+	"os"
+	"regexp"
+	"syscall"
+)
+
+// Volume Path aware implementation of MkdirAll for Windows.
+func MkdirAll(path string, perm os.FileMode) error {
+	re := regexp.MustCompile("^\\\\\\\\\\?\\\\Volume{[a-z0-9-]+}$")
+	if re.MatchString(path) {
+		return nil
+	}
+
+	// Fast path: if we can tell whether path is a directory or file, stop with success or error.
+	dir, err := os.Stat(path)
+	if err == nil {
+		if dir.IsDir() {
+			return nil
+		}
+		return &os.PathError{
+			Op:   "mkdir",
+			Path: path,
+			Err:  syscall.ENOTDIR,
+		}
+	}
+
+	// Slow path: make sure parent exists and then call Mkdir for path.
+	i := len(path)
+	for i > 0 && os.IsPathSeparator(path[i-1]) { // Skip trailing path separator.
+		i--
+	}
+
+	j := i
+	for j > 0 && !os.IsPathSeparator(path[j-1]) { // Scan backward over element.
+		j--
+	}
```

Comment:
Can we simplify those handcrafted loops with [`filepath.SplitList`](http://golang.org/pkg/path/filepath/#SplitList)?